classes/NewsFetcher.php

```diff
@@ -0,0 +1,154 @@
+            ->setStrippedTimeout(self::OPEN_TIMEOUT_SECONDS)
+            ->setStorage($storage)
+            ->setClientOptions([
+                'method' => 'GET',
+                'subscribers' => [$cacheSubscriber],
+            ])
+        ;
+        $this->factory = new AdvancedCircuitBreakerFactory();
+    }
+
+    /**
+     * @return array
+     *
+     * @throws PrestaShopException
+     */
+    public function getData()
+    {
+        $data = ['has_errors' => true, 'rss' => []];
+
+        $circuitBreaker = $this->factory->create($this->blogSettings);
+        $blogXMLResponse = $circuitBreaker->call(_PS_API_URL_ . '/rss/blog/blog-' . $this->isoCode . '.xml');
+
+        if (empty($blogXMLResponse)) {
+            return $data;
+        }
+
+        $rss = @simplexml_load_string($blogXMLResponse);
+        if (!$rss) {
+            return $data;
+        }
+
+        $articles_limit = self::NUM_ARTICLES;
+
+        $shop_default_country_id = (int) Configuration::get('PS_COUNTRY_DEFAULT');
+        $shop_default_iso_country = (string) Tools::strtoupper(Country::getIsoById($shop_default_country_id));
+        $analytics_params = [
+            'utm_source' => 'back-office',
+            'utm_medium' => 'rss',
+            'utm_campaign' => 'back-office-' . $shop_default_iso_country,
+        ];
+
+        foreach ($rss->channel->item as $item) {
+            if ($articles_limit == 0
+                || !Validate::isCleanHtml((string) $item->title)
+                || !Validate::isCleanHtml((string) $item->description)
+                || !isset($item->link, $item->title)) {
+                break;
+            }
+            $analytics_params['utm_content'] = 'download';
+            if (in_array($this->context->mode, array(Context::MODE_HOST, Context::MODE_HOST_CONTRIB))) {
+                $analytics_params['utm_content'] = 'cloud';
+            }
+            $article_link = (string) $item->link . '?' . http_build_query($analytics_params);
+
+            $url_query = parse_url($item->link, PHP_URL_QUERY);
+            parse_str($url_query, $link_query_params);
+            if ($link_query_params) {
+                $full_url_params = array_merge($link_query_params, $analytics_params);
+                $base_url = explode('?', (string) $item->link);
+                $base_url = (string) $base_url[0];
+                $article_link = $base_url . '?' . http_build_query($full_url_params);
+            }
+
+            $data['rss'][] = [
+                'date' => Tools::displayDate(date('Y-m-d', strtotime((string) $item->pubDate))),
+                'title' => (string) Tools::htmlentitiesUTF8($item->title),
+                'short_desc' => Tools::truncateString(strip_tags((string) $item->description), 150),
+                'link' => (string) $article_link,
+            ];
+            --$articles_limit;
+        }
+        $data['has_errors'] = false;
+
+        return $data;
+    }
+}
```

Comment:
you can use PHP CS Fixer to add a EOF line

Reply:
:+1: